src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -617,44 +617,47 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
         {
             try
             {
-                // do not send invocation requests for functions that failed to load or could not be indexed by the worker
+                var invocationId = context.ExecutionContext.InvocationId.ToString();
+
+                // do not send an invocation request for functions that failed to load or could not be indexed by the worker
                 if (_functionLoadErrors.ContainsKey(context.FunctionMetadata.GetFunctionId()))
                 {
-                    _workerChannelLogger.LogDebug($"Function {context.FunctionMetadata.Name} failed to load");
+                    _workerChannelLogger.LogDebug("Function {functionName} failed to load", context.FunctionMetadata.Name);
                     context.ResultSource.TrySetException(_functionLoadErrors[context.FunctionMetadata.GetFunctionId()]);
-                    _executingInvocations.TryRemove(context.ExecutionContext.InvocationId.ToString(), out ScriptInvocationContext _);
+                    _executingInvocations.TryRemove(invocationId, out ScriptInvocationContext _);
                     return;
                 }
                 else if (_metadataRequestErrors.ContainsKey(context.FunctionMetadata.GetFunctionId()))
                 {
-                    _workerChannelLogger.LogDebug($"Worker failed to load metadata for {context.FunctionMetadata.Name}");
+                    _workerChannelLogger.LogDebug("Worker failed to load metadata for {functionName}", context.FunctionMetadata.Name);
                     context.ResultSource.TrySetException(_metadataRequestErrors[context.FunctionMetadata.GetFunctionId()]);
-                    _executingInvocations.TryRemove(context.ExecutionContext.InvocationId.ToString(), out ScriptInvocationContext _);
+                    _executingInvocations.TryRemove(invocationId, out ScriptInvocationContext _);
                     return;
                 }
 
+                // do not send an invocation request if cancellation has been requested
                 if (context.CancellationToken.IsCancellationRequested)
                 {
-                    _workerChannelLogger.LogDebug("Cancellation has been requested, cancelling invocation request");
-                    context.ResultSource.SetCanceled();
+                    _workerChannelLogger.LogWarning("Cancellation has been requested. The invocation request with id '{invocationId}' is cancelled and will not be sent to the worker.", invocationId);
+                    context.ResultSource.TrySetCanceled();
```

Comment:
Do you want to do anything with the result of `TrySetCanceled()`? ex. log a message if it returns false, which shouldn't happen in with the changes you're making here? (And thus, should be checked for because trust nothing.)

Reply:
I don't think we need any additional logs or checks here. The try is just in case the resultsource has already been disposed or cancelled, we protect ourselves from an exception or other issues.